Un-hardcode target cursors for various traits

dragunoff opened the pull request:
As mentioned in #18087 the target cursor for `DeliversCash` is not configurable. I set out to fix that but noticed that there are many more hard-coded cursors in various traits so I took a pass at all of them.

The last commit before the testcase is just formatting because the changes to `EnterAlliedActorTargeter` resulted in very long argument lists.

The testcase changes all target cursors of affected traits to `goldwrench` and `goldwrench-blocked`.

dragunoff committed:
Add configurable target cursors to various traits

dragunoff committed:
Add configurable cursors for entering allied actor targeters

dragunoff committed:
Formatting: wrap and indent long argument lists

tovl commented on OpenRA.Mods.Common/Traits/Air/Aircraft.cs:
```diff
@@ -139,6 +139,9 @@ public class AircraftInfo : PausableConditionalTraitInfo, IPositionableInfo, IFa
 		[Desc("Boolean expression defining the condition under which the regular (non-force) move cursor is disabled.")]
 		public readonly BooleanExpression RequireForceMoveCondition = null;
 
+		public readonly string EnterCursor = "enter";
+		public readonly string EnterBlockedCursor = "enter-blocked";
```
Shouldn't we add descriptions for these? Same for the other traits here.

dragunoff committed:
Add missing trait descriptions for cursors and unify the language

dragunoff commented:
Update: Added descriptions to all cursor fields and unified the language used across them. The descriptions started with _"Cursor to display|to show|to use|sequence to use"_ and the most widely used was _"Cursor to display..."_ so I changed all of them to use this language (even though _"Cursor sequence to use..."_ is more accurate).

There are still a few places with hard-coded cursors (harvesting and mine-laying). I could take care of them in this PR or leave to a follow up.

tovl reviewed (dismissed):
:+1:  after the testcases are removed. I am not a fan of the one-line-per-argument thing (I would've just broken the line into two and pulled the lambdas into local variables) but that is just a personal preference; I don't think we have any style guide for how lines that are too long should be broken up (or what is too long in the first place).

dragunoff commented:
Rebased. Will remove the testcase after +2.

abcdefg30 reviewed (dismissed):
Fwiw, also not a fan of spreading onto so many lines, but I guess it is now consistent with the other existing ones. Testcase can be removed now.

dragunoff commented:
Update: removed testcase and fixed conflicts.

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/755caef35d8725178e97585df1c8072fac75979f...726fb136af004889c5df28a4ffcd3fb999e2ac9f)
